### PR TITLE
add workflow to tag docs release on repo release

### PR DESCRIPTION
**Pull request by fg-j**
## Summary

This workflow runs when a release is cut on the repo. It creates a new tag `docs/vX.Y.Z` where `vX.Y.Z` is the version of the release that's cut on the repo. Because Hugo modules requires that the `docs` subdirectory have its own `go.mod`, that makes the subdirectory a go submodule. Tags prefixed with `docs/` are the version tags for the submodule. Since the `docs/` directory isn't just a package, its versioning does not come directly from the repo's main tags. (See https://github.com/go-modules-by-example/index/blob/master/009_submodules/README.md) for an explanation of Go submodules and their versioning.

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.

### CODE REVIEW

**Comment by fg-j** [comment]
Blocked until paketo-buildpacks/github-config#258 merges.

**Review by ForestEckhardt — APPROVED** [review]

**Pull request merged**
